src/SparkFun_AS7331.h

```diff
@@ -14,1312 +14,716 @@
-        /// @return 0 if successful, negative if error, positive for warning.
-        int8_t getOptIndex(sfe_as7331_reg_cfg_optreg_t *optReg)
-        {
-            return readRegister(SFE_AS7331_REGISTER_CFG_OPTREG, &optReg->byte);
-        }
-
-        /// @brief Sets the option register when in configuration operation mode.
-        /// @param optReg Pointer to a register struct that has the new register configuration.
-        /// @return 0 if successful, negative if error, positive for warning.
-        int8_t setOptIndex(const sfe_as7331_reg_cfg_optreg_t *optReg)
-        {
-            return writeRegister(SFE_AS7331_REGISTER_CFG_OPTREG, &optReg->byte);
-        }
-
-        sfeAS7331_measures_t measures;
-
-    private:
-        /// @brief Converts the raw temperature value to a human readable form.
-        /// @param inputVal Raw temperature value to convert.
-        /// @return The converted device temperature in degree C.
-        float convertRawTempToTempC(const uint16_t inputVal)
-        {
-            // T_chip = TEMP*0.05 - 66.9
-            // EX: TEMP=0x922 aka TEMP=0d2338, returns 50.0
-            return ((float)(inputVal) * 0.05f) - 66.9f;
-        }
-
-        /// @brief Called when changing values that affect the conversion, calculates a new conversion factor to reduce the conversion overhead.
-        void calculateConversionFactors(void)
-        {
-            float divFactor = (bool)_config.dividerEnabled ? (float)(1 << (1+_config.dividerRange)) : 1.0f;
-            _conversionA = fsrA/((float)(1 << (11 - _config.sensorGain)))/((float)(1 << _config.conversionTime))/((float)(1024.0 * (1 << _config.cclk)))*divFactor;
-            _conversionB = fsrB/((float)(1 << (11 - _config.sensorGain)))/((float)(1 << _config.conversionTime))/((float)(1024.0 * (1 << _config.cclk)))*divFactor;
-            _conversionC = fsrC/((float)(1 << (11 - _config.sensorGain)))/((float)(1 << _config.conversionTime))/((float)(1024.0 * (1 << _config.cclk)))*divFactor;
-        }
-
-        sfeBusDevice *_sfeBus; // Pointer to bus device.
-        sfeBusDeviceSettings *_devSettings; // Pointer to the device settings for this sensor.
-
-        sfeAS7331_config_t _config; // Internal sensor configuration.
-        sfeAS7331_state_t _state; // Internal sensor state.
-
-        float _conversionA; // Internal conversion factor for the UVA sensor.
-        float _conversionB; // Internal conversion factor for the UVB sensor.
-        float _conversionC; // Internal conversion factor for the UVC sensor.
-
-        static constexpr float fsrA = 348160.0; // Full Scale Resolution for the UVA sensor.
-        static constexpr float fsrB = 387072.0; // Full Scale Resolution for the UVB sensor.
-        static constexpr float fsrC = 169984.0; // Full Scale Resolution for the UVC sensor.
+    /// @brief Checks to see if the AS7331 is connected.
+    /// @return True if successful, false otherwise.
+    bool isConnected(void)
```

Comment:
In this specific instance, sure. However, from a standardization perspective this should exist in the interface-specific implementation. Since this is supposed to be one of the first examples for proper implementation of the Toolkit library, we should do it correctly for a generic implementation.